### PR TITLE
Support app2app

### PR DESCRIPTION
**Pull request by tung2744**
ref #138

### CODE REVIEW

File: Sources/app2app/App2App.swift [hunk 1]
```diff
@@ -33,49 +33,70 @@ class App2App {
...
+        }
+        return privateKey
+    }
+    
     @available(iOS 11.3, *)
     private func generatePrivateKey(tag: String) throws -> SecKey {
+        do {
+            return try generatePrivateKeyInSecureEnclave(tag: tag)
+        } catch {
+            // Fallback if generation in secure enclave is failed for any reason
```
[the remark this answers is not in the document]
Comment (tung2744) [reply Sources/app2app/App2App.swift:59]: @louischan-oursky I think not all device support using secure enclave. If the device does not support it we may need to fallback so he can still use app2app.

[the remark this answers is not in the document]
Comment (louischan-oursky) [reply Sources/app2app/App2App.swift:59]: https://developer.apple.com/documentation/security/certificate_key_and_trust_services/keys/protecting_keys_with_the_secure_enclave

> Requires hardware support. Only iOS devices with an A7 or later processor, or a Mac with the Touch Bar and Touch ID or with an M1 or later processor support this feature.

And A7 chip is from 2013 https://en.wikipedia.org/wiki/Apple_A7 Given our support for iOS version is latest 3 versions, we can assume all devices we support can use secure enclave.

[the remark this answers is not in the document]
Comment (tung2744) [reply Sources/app2app/App2App.swift:59]: ok, then let me remove the fallback.

**Comment by tung2744** [comment]
@louischan-oursky Updated, thanks!

**Comment by louischan-oursky** [comment]
@tung2744 In this PR we have vendored some external source files. They were formatted by swiftformat. This is undesirable. Can you revert the format we have done on the source files, and add an exclusion in swiftformat to exclude those files? https://github.com/nicklockwood/SwiftFormat#config-file

Also please make sure we always place vendored code in their own files and exclude them in swiftformat.

**Comment by tung2744** [comment]
@louischan-oursky Updated to extract and ignore external source files. Also removed redundant internals from our source code.

**Comment by louischan-oursky** [comment]
Merged in https://github.com/authgear/authgear-sdk-ios/commit/f63c467a55a6e8f435a3cd3044a2a6412b130ed7

**Pull request closed without merging**